src/Features/CSharp/Portable/SignatureHelp/PrimaryConstructorBaseTypeSignatureHelpProvider.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.DocumentationComments;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SignatureHelp;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.SignatureHelp
+{
+    /// <summary>
+    /// Implements SignatureHelp and ParameterInfo for <see cref="PrimaryConstructorBaseTypeSyntax"/>
+    /// such as 'record Student(int Id) : Person($$"first", "last");`.
+    /// </summary>
+    [ExportSignatureHelpProvider("PrimaryConstructorBaseTypeSignatureHelpProvider", LanguageNames.CSharp), Shared]
+    internal partial class PrimaryConstructorBaseTypeSignatureHelpProvider : AbstractCSharpSignatureHelpProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public PrimaryConstructorBaseTypeSignatureHelpProvider()
+        {
+        }
+
+        public override bool IsTriggerCharacter(char ch)
+            => ch == '(' || ch == ',';
+
+        public override bool IsRetriggerCharacter(char ch)
+            => ch == ')';
+
+        private bool TryGetBaseTypeSyntax(SyntaxNode root, int position, ISyntaxFactsService syntaxFacts, SignatureHelpTriggerReason triggerReason, CancellationToken cancellationToken, out PrimaryConstructorBaseTypeSyntax expression)
+        {
+            if (!CommonSignatureHelpUtilities.TryGetSyntax(root, position, syntaxFacts, triggerReason, IsTriggerToken, IsArgumentListToken, cancellationToken, out expression))
+            {
+                return false;
+            }
+
+            return expression.ArgumentList != null;
+
+            static bool IsArgumentListToken(PrimaryConstructorBaseTypeSyntax expression, SyntaxToken token)
+            {
+                return expression.ArgumentList != null &&
+                    expression.ArgumentList.Span.Contains(token.SpanStart) &&
+                    token != expression.ArgumentList.CloseParenToken;
```

Comment:
is this generally how this funciton is written.  i.e. the token doesn't have to be a direct child of the ArgList, but just is within it's span?  (if this matches other helps, cool, if not an explanation woudl help).

Reply:
Yes, this method was unmodified (from `ConstructorInitializerSignatureHelpProvider`)

Reply:
thanks.